src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TargetElementAttribute.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.Razor.TagHelpers;
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    /// <summary>
+    /// Used to override a <see cref="ITagHelper"/>'s default element target.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = true, Inherited = false)]
+    public sealed class TargetElementAttribute : Attribute
```

Comment:
- not sure `[TargetElement]` makes sense given support for attribute-only targets
- may want to make the attribute name more specific to tag helpers and align with `[HtmlAttributeName]` e.g. `[HtmlTarget]`